### PR TITLE
Remove all remaining MyCodeAction types in the repo

### PR DESCRIPTION
**Pull request by mavasani**
Follow up to #60480 and #60557

### CODE REVIEW

**Comment by mavasani** [comment]
@Youssef1313 for review

File: src/Features/CSharp/Portable/ConvertBetweenRegularAndVerbatimString/AbstractConvertBetweenRegularAndVerbatimStringCodeRefactoringProvider.cs [hunk 1]
```diff
@@ -53,20 +53,28 @@ public sealed override async Task ComputeRefactoringsAsync(CodeRefactoringContex
...
 
+            // Note: This is a generally useful feature on strings.  But it's not likely to be something
+            // people want to use a lot.  Make low priority so it doesn't interfere with more
+            // commonly useful refactorings.
+
             if (IsVerbatim(literalExpression))
             {
                 // always offer to convert from verbatim string to normal string.
-                context.RegisterRefactoring(new MyCodeAction(
+                context.RegisterRefactoring(CodeAction.CreateWithPriority(
```
Comment (mavasani) [inline src/Features/CSharp/Portable/ConvertBetweenRegularAndVerbatimString/AbstractConvertBetweenRegularAndVerbatimStringCodeRefactoringProvider.cs:63]: I added a new internal helper `CodeAction.CreateWithPriority` for creating code actions with non-default priority. We no longer need to create special code action sub-types just to override the priority.

**Review by Youssef1313 — APPROVED** [review]
Few missing code actions that were not named "MyCodeAction"

- `PreferFrameworkTypeCodeFixProvider` (`PreferFrameworkTypeCodeAction`)
- `AbstractConvertAutoPropertyToFullPropertyCodeRefactoringProvider` (`ConvertAutoPropertyToFullPropertyCodeAction`)
- `AbstractConvertForEachToLinqQueryProvider` (`ForEachToLinqQueryCodeAction`)
- `AbstractConvertPlaceholderToInterpolatedStringRefactoringProvider` (`ConvertToInterpolatedStringCodeAction`)
- `AbstractConvertForEachToForCodeRefactoringProvider` (`ForEachToForCodeAction`)

Other than that, LGTM 🎉

**Review by Youssef1313 — COMMENTED** [review]
I'm confused regarding the `RegisterCodeFix(context, ...` introduced in #60665 vs `context.RegisterCodeFix(...)`.

Which should be used?

**Comment by mavasani** [comment]
> I'm confused regarding the `RegisterCodeFix(context, ...` introduced in #60665 vs `context.RegisterCodeFix(...)`.
> 
> Which should be used?

I believe Tomas just introduced a helper in that PR, so if the fund you were passing to RegisterCodeFix only invokes the base type’s FixAsync method, then you should use the new helper to reduce verbosity.

**Comment by mavasani** [comment]
Thanks @Youssef1313, that helped! I actually found a few more redundant sub-types, all of them should be removed with the latest commit.

**Review by Youssef1313 — APPROVED** [review]

**Review by jmarolf — APPROVED** [review]
Thanks for all the work here! Do we have tests for code action priority that ensure we are laying things out in the right order?

**Comment by mavasani** [comment]
> Thanks for all the work here! Do we have tests for code action priority that ensure we are laying things out in the right order?

I think we have for some, not for all. Let me create an issue to track adding the tests.

**Comment by mavasani** [comment]
Filed https://github.com/dotnet/roslyn/issues/61081

**Pull request merged**